File: graphql-api-java/demo/src/main/java/com/graphql/demo/resolver/PersonResolver.java
package com.graphql.demo.resolver;

import com.coxautodev.graphql.tools.GraphQLMutationResolver;
import com.coxautodev.graphql.tools.GraphQLQueryResolver;
import com.graphql.demo.input.PersonInput;
import com.graphql.demo.model.Person;
import com.graphql.demo.model.Playlist;
import com.graphql.demo.repository.PersonRepository;
import com.graphql.demo.repository.PlaylistRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

@Component
public class PersonResolver implements GraphQLQueryResolver, GraphQLMutationResolver {

    @Autowired
    private PersonRepository repository;
    @Autowired
    private PlaylistRepository playlistRepository;

    public Collection<Person> findAllPeople(){
        return repository.findAll();
    }

    public Person savePerson(PersonInput input){
        return repository.save(new Person(input.getName(), input.getAge()));
    }

    public Person findPersonByName(String name){
        for(Person p:repository.findAll()){
            if(p.getName().equals(name)){
                return p;
            }
        }
        return null;
    }
}

File: graphql-api/demo/src/main/java/com/graphql/demo/resolver/PersonResolver.kt
package com.graphql.demo.resolver

import com.coxautodev.graphql.tools.GraphQLMutationResolver
import com.coxautodev.graphql.tools.GraphQLQueryResolver
import com.graphql.demo.input.PersonInput
import com.graphql.demo.model.Person
import com.graphql.demo.repository.PersonRepository
import com.graphql.demo.repository.PlaylistRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class PersonResolver : GraphQLQueryResolver, GraphQLMutationResolver {
    @Autowired
    private val repository: PersonRepository? = null

    @Autowired
    private val playlistRepository: PlaylistRepository? = null
    fun findAllPeople(): Collection<Person?> {
        return repository!!.findAll()
    }

    fun savePerson(input: PersonInput): Person {
//        return repository!!.save(Person(input.name!!, input.getAge()))
        return repository!!.save(Person(input.name!!, input.age!!))
    }

    fun findPersonByName(name: String?): Person? {
        for (p in repository!!.findAll()) {
//            if (p.getName().equals(name)) {
            if (p != null) {
                if (p.name == name) {
                    return p
                }
            }
        }
        return null
    }
}